CDCgov/cdc-open-viz
# dev-8412 API Nested Dropdown

## DEV-8412

Added the URL option as a data source for Nested Dropdowns 

## Testing Steps

Scenario: Open a Default Editor and choose the Dashboard Viz and in Configuration, upload dev-8412-Nested-Dropdown.json (attached to ticket, [https://websupport.cdc.gov/browse/DEV-8412] )

Expected: There are 2 Visualization boxes. The one on top is empty and the bottom one has a table in it.

Scenario: Click on the Impot Data Tab and select Load from URL

1. Click the "Always load from URL" checkbox
2. Enter `https://nccd-cove-public-api.apps.ecpaas-dev.cdc.gov/od-public?$datakey=brfss_prevalence_cove_explore_by_topic`
3. Press the Load button

Expected: Data should populate on the right hand side under Data Preview

4. Click the Configure tab
5. Drag a Filter Dropdowns viz into the Visualization area
6. Click on the Gear icon for the Table
7. In Select a dataset box select the URL, vertical, no, and then click continue
8. Click the tools icon of the Filter Dropdowns viz

Expected: Filters Dropdown Viz will open with a Configure Dashboard Filter on the left and a blank screen on the right

9. Click the Filter Accordion
10. Click Add Filter
11. Click the Down arrow button
12. Select URL in the Filter Type dropdown
13. Select Nested Dropdown in the Filter Style dropdown
14. In the API Endpoint text box, add `https://nccd-cove-public-api.apps.ecpaas-dev.cdc.gov/od-public?$datakey=brfss_prevalence_cove_explore_by_topic&$select=distinct%20Class,Topic`
15. In Value Selector enter Class
16. In Subgroup Value Selector enter Topic
17. In Used By select table1727…
18. Click on the filter that was populated on the right

Expected: A nested dropdown list will appear with Class as the bold group titles and the Topic will be the options under the group titles.

Scenario: Click on the Back to Dashboard button and click the Dashboard Preview tab

Expected: There will be the Dashboard filter on top with a Table below with data already populated.

19. Click on the Filter
20. Select Binge Drinking under Alcohol Consumption title

Expected: The Table will be filtered to just show Binge Drinking data

## Self Review

- I have added testing steps for reviewers
- My changes generate same key error when the Group is populated in the dropdown and the subgroup hasn't populated yet. Once the subgroup is populated then there are no same key errors.
